tests/Tests/SKCodecTest.cs

```diff
@@ -491,5 +493,31 @@ public void CanDecodeData(string image)
 			Assert.Equal(SKCodecResult.Success, codec.GetPixels(out var pixels));
 			Assert.NotEmpty(pixels);
 		}
+
+		[SkippableFact]
+		public void CanReadPngChunks_iTXt()
+		{
+			var path = Path.Combine(PathToImages, "png_chunks/good_itxt.png");
+
+			using var chunkReader = new TestPngChunkReader();
+			using var codec = SKCodec.Create(path, chunkReader, SKCodecSelectionPolicy.PreferStillImage, out var codecResult);
+			using var bmp = SKBitmap.Decode(codec);
+
+			var chunks = chunkReader.Chunks;
+
+			Assert.NotEmpty(chunks);
+		}
```

Comment:
This test is failing because chunks are not being read still.

Reply:
could you try the following?

```cs
internal class Reader : SKPngChunkReader
{
    public bool r = false;

    protected override bool ReadChunk(string tag, IntPtr data, IntPtr length)
    {
        r = true;
        return true;
    }
}
public void CanInvokePngReadChunks(
{
    using Reader chunkReader = new();

    var path = Path.Combine(PathToImages, "png_chunks/good_itxt.png");

    using var codec = SKCodec.Create(path, chunkReader, SKCodecSelectionPolicy.PreferStillImage, out var codecResult);

    Assert.True(r.r, "failed to invoke chunk reader");
}
```

with `SkiaSharp\externals\skia\BUILD.gn`

```
optional("png_decode") {
  enabled = skia_use_libpng_decode
  public_defines = [ "SK_CODEC_DECODES_PNG", "PNG_READ_UNKNOWN_CHUNKS_SUPPORTED" ]

  deps = [ "//third_party/libpng" ]
  sources = [
    "src/codec/SkIcoCodec.cpp",
    "src/codec/SkPngCodec.cpp",
  ]
}
```

Reply:
I already set the flag here https://github.com/mono/SkiaSharp/pull/2149#discussion_r946167028

I can actually debug and see it reading the chunk but it seems to decide it is not worth it and skips.

Reply:
oof

Reply:
https://groups.google.com/g/skia-discuss/c/zHVOjwoQ-jQ/m/aKQ-EIZ_AQAJ

it looks like we will need to test it with UNKNOWN chunks instead of KNOWN chunks

Reply:
coincidentally this `[an example implementation here](https://android.googlesource.com/platform/frameworks/base/+/master/libs/hwui/jni/NinePatchPeeker.cpp).` is exactly what i have ported https://github.com/mgood7123/AndroidUI/blob/master/AndroidUI/Graphics/NinePatchPeeker.cs

Reply:
but i currently dont have any NinePatch images to test with, however the NinePatch tests all pass https://github.com/mgood7123/android_9_patch/blob/31931f9910f428207ee482a808cc09195a985c87/9patch_tests.cpp

Reply:
and https://github.com/mgood7123/AndroidUI/blob/f7746e01b17bd728f94937da5d96c34c5d6ea024/AndroidUI/Graphics/BitmapFactory.cs#L1254

Reply:
I wonder if we can borrow some images from android in the GitHub.com/aosp_mirror